### PR TITLE
[pipeline] add pipeline tests to ci

### PR DESCRIPTION
**Pull request by jigold**
@cseed I tried following the structure from the batch tests. This PR should go in before #5962.

### CODE REVIEW

**Comment by akotlar** [comment]
checking

File: pipeline/Dockerfile.test [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM {{ base_image.image }}
+
+COPY hailjwt/setup.py /hailjwt/
+COPY hailjwt/hailjwt /hailjwt/hailjwt/
+RUN pip3 install --no-cache-dir /hailjwt && \
+  rm -rf /hailjwt
+
+COPY batch/setup.py /batch/
+COPY batch/batch/ /batch/batch/
```
Comment (akotlar) [inline pipeline/Dockerfile.test]: Forward slash on batch/batch/ (src) is not needed:

https://docs.docker.com/engine/reference/builder/

"""
If <src> is a directory, the entire contents of the directory are copied, including filesystem metadata.

Note: The directory itself is not copied, just its contents.
"""

File: pipeline/Dockerfile.test [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+RUN pip3 install --no-cache-dir /batch && \
+  rm -rf /batch
+
+COPY pipeline/setup.py /pipeline/
+COPY pipeline/pipeline/ /pipeline/pipeline/
```
Comment (akotlar) [inline pipeline/Dockerfile.test]: same

File: pipeline/Dockerfile.test [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+RUN pip3 install --no-cache-dir /pipeline && \
```
Comment (akotlar) [inline pipeline/Dockerfile.test]: This confuses me. We're copying contents of `pipeline/pipeline to /pipeline/pipeline/`. However, in the next step we rm -rf /pipeline. Isn't the net effect that this copy step does nothing?

File: pipeline/Dockerfile.test [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  rm -rf /pipeline
+
+COPY pipeline/test/ /test/
```
Comment (akotlar) [inline pipeline/Dockerfile.test]: same, no trailing slash needed in `pipeline/test/`

File: pipeline/Dockerfile.test [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+COPY hailjwt/setup.py /hailjwt/
+COPY hailjwt/hailjwt /hailjwt/hailjwt/
+RUN pip3 install --no-cache-dir /hailjwt && \
+  rm -rf /hailjwt
+
+COPY batch/setup.py /batch/
+COPY batch/batch/ /batch/batch/
+RUN pip3 install --no-cache-dir /batch && \
+  rm -rf /batch
```
Comment (akotlar) [inline pipeline/Dockerfile.test]: Same as below comment, /batch appears to be removed, making the 2nd copy step unnecessary.

**Comment by jigold** [comment]
Closing this and including it with a different PR.

**Pull request closed without merging**